### PR TITLE
Merge split-crates branch into main

### PR DESCRIPTION
**Pull request by zeskeertwee**

### CODE REVIEW

**Comment by zeskeertwee** [comment]
i'm waiting for this PR to go through before i create a PR to merge the remove-mime-type branch

File: virtfs-ach/src/tests/writer.rs [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+const FILE_1: &[u8] = include_bytes!("../../test-data/data.bin");
+const FILE_1_PATH: &str = "data.bin";
+const FILE_2: &[u8] = include_bytes!("../../test-data/lorem_ipsum.txt");
+const FILE_2_PATH: &str = "lorem_ipsum.txt";
 
 #[test]
 fn archive_writing() {
     let mut builder = ArchiveBuilder::new();
     builder.add_file(
         FILE_1.to_owned(),
```
Comment (sokorototo) [inline virtfs-ach/src/tests/writer.rs:15]: Maybe just use a dereference here

Comment (sokorototo) [inline virtfs-ach/src/tests/writer.rs:15]: Maybe just dereference here

**Review by sokorototo — COMMENTED** [review]
Seems okay to me. But I thought `vf` as a crate name was short and to the point: "Virtual Filesystem", maybe change the name

File: virtfs-ach/src/tests/writer.rs (hunk 1 above)
Comment (zeskeertwee) [reply virtfs-ach/src/tests/writer.rs:15]: rust won't let me dereference that without setting the exact size, which is really impractical

**Comment by zeskeertwee** [comment]
oh we can definitly change the name from `virtfs` to `vf` later

**Comment by sokorototo** [comment]
Yeah sure

**Pull request merged**